### ISSUE TITLE
False positive NUnit2020 for Has.None.SameAs

### ISSUE THREAD

**Issue by jnm2** [nunit/nunit.analyzers#230]
This warning would be correct for `Is.SameAs`, but it is not taking into account the `Has.None` part.

```cs
class C
{
    [Test]
    public void M()
    {
        // NUnit2020 Provided actual and expected arguments for SameAs constraint cannot have
        // same type.                                   ↓↓↓↓↓↓↓↓↓↓↓↓
        Assert.That(new List<string>(), Has.None.SameAs(string.Empty));
    }
}
```

Nitpicking ahead, feel free to take it or leave it:

The message should be more casual and accessible. Right now, it is terse to the point of detracting from its readability. There are no articles (a/an/the) in the sentence. It is a good practice though to look for words to delete in every message without losing meaning or readability. For example, I think that the word 'provided' could be removed in the message above without loss of clarity. Here's a placeholder example: "The SameAs constraint always fails because the actual and expected values have mutually exclusive types."

A comprehensive resource that helps me is https://docs.microsoft.com/en-us/style-guide/top-10-tips-style-voice. I like what they're doing.

**Comment by mikkelbu** [comment]
I think I'll move the part about the message to a separate issue (I will create it tomorrow) - as we need to take a look at all the title, messages, and descriptions.

**Comment by Dreamescaper** [comment]
Adding proper support for All/None/etc operators probably depends on https://github.com/nunit/nunit.analyzers/issues/218.
As a quick fix we can allow only `Not` operator for the analyzer, as it is done in several other analyzers.

**Comment by mikkelbu** [comment]
I think it is fine to go with the quick fix not and we can improve on this later.
